pyne/mesh.py

```diff
@@ -443,23 +513,38 @@ def __delitem__(self, key):
         size = len(self.mesh)
         mtag = self.tag
         miter = self.mesh.iter_ve()
-        if isinstance(key, _INTEGRAL_TYPES):
+        if isinstance(key, Mesh) and key == self.mesh: # special case, look up mesh's root set
+            self.mesh.mesh.tag_delete_data(mtag,
+                                           self.mesh.mesh.get_root_set())
+        elif isinstance(key, _eh_py_type):
+            self.mesh.mesh.tag_delete_data(mtag, key)
+        elif isinstance(key, _INTEGRAL_TYPES):
             if key >= size:
                 raise IndexError("key index {0} greater than the size of the "
                                  "mesh {1}".format(key, size))
             for i_ve in zip(range(key+1), miter):
                 pass
-            del mtag[i_ve[1]]
+            self.mesh.mesh.tag_delete_data(mtag, i_ve[1])
         elif isinstance(key, slice):
-            del mtag[list(miter)[key]]
+            self.mesh.mesh.tag_delete_data(mtag, list(miter)[key])
         elif isinstance(key, np.ndarray) and key.dtype == np.bool:
             if len(key) != size:
                 raise KeyError("boolean mask must match the "
                                "length of the mesh.")
-            del mtag[[ve for b, ve in zip(key, miter) if b]]
+            self.mesh.mesh.tag_delete_data(mtag,[ve for b, ve in zip(key, miter) if b])
         elif isinstance(key, Iterable):
             ves = list(miter)
-            del mtag[[ves[i] for i in key]]
+            ves_to_delt = []
```

Comment:
```suggestion
            ves_to_del = []
```
coverage: maybe this isn't tested?

Reply:
addressed in #59